File: src/main/java/site/gachontable/gachontablebe/domain/waiting/usecase/CancelWaiting.java
package site.gachontable.gachontablebe.domain.waiting.usecase;

import site.gachontable.gachontablebe.domain.waiting.presentation.dto.request.CancelRequest;
import site.gachontable.gachontablebe.domain.waiting.presentation.dto.response.WaitingResponse;

public interface CancelWaiting {
    WaitingResponse execute(CancelRequest request);
}

File: src/main/java/site/gachontable/domain/waiting/port/in/CancelWaiting.kt
package site.gachontable.domain.waiting.port.`in`

import site.gachontable.presentation.waiting.dto.request.CancelRequest
import site.gachontable.presentation.waiting.dto.response.WaitingResponse

interface CancelWaiting {
    fun execute(request: CancelRequest, lockKey: String): WaitingResponse
}
